api/jobs/handlers.py

```diff
@@ -479,6 +481,51 @@ def retry(self, _id):
         new_id = Queue.retry(j, force=True)
         return { "_id": new_id }
 
+
+    @require_drone
+    def prepare_complete(self, _id):
+        j = Job.get(_id)
+        payload = self.request.json
+        ticket = {
+            'job': j.id_,
+            'success': payload.get('success', False),
+        }
+        return {'ticket': config.db.job_tickets.insert_one(ticket).inserted_id}
+
+
+    @require_login
+    def accept_failed_output(self, _id):
+        j = Job.get(_id)
+
+        # Permission check
+        if not self.superuser_request:
+            j.destination.check_access(self.uid, 'rw')
+
+        if j.state != 'failed':
+            self.abort(400, 'Can only accept failed output of a job that failed')
+
+        # Remove flag from files
+        container = j.destination.get()
+        container_before = copy.deepcopy(container)
+        for f in container.get('files'):
+            if f['origin'] == {'type': 'job', 'id': _id}:
+                del f['from_failed_job']
```

Comment:
I notice there's no access logging for this change. @nagem, do you think we should record who decided to go ahead & mark failed job data as accepted? Seems like an important decision that should be recorded.